beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/aws/S3Client.java

```diff
@@ -73,10 +74,20 @@ List<String> deleteObjects(String bucket, List<String> keys) {
       return Collections.emptyList();
     }
     if (!dryRunEnabled) {
-      keys.forEach(key -> log.info("Deleting: \"{}/{}\"", bucket, key));
-      DeleteObjectsRequest deleteObjectsRequest = new DeleteObjectsRequest(bucket)
-          .withKeys(keys.toArray(new String[] {}));
-      DeleteObjectsResult deleteObjectsResult = amazonS3.deleteObjects(deleteObjectsRequest);
+      log.info("Attempting to delete a total of {} objects, from [{}] to [{}]", keys.size(), keys.get(0), keys.get(keys.size() - 1));
+      DeleteObjectsResult deleteObjectsResult = new DeleteObjectsResult(new ArrayList<>());
+      DeleteObjectsRequest deleteObjectsRequest = new DeleteObjectsRequest(bucket);
+      int totalKeys = keys.size();
+      int indexStart;
+      int indexEnd = 0;
+      while(indexEnd < totalKeys) {
+        indexStart = indexEnd;
+        indexEnd = nextIndexEnd(indexStart, REQUEST_CHUNK_SIZE, totalKeys);
+        deleteObjectsRequest.withKeys(keys.subList(indexStart, indexEnd).toArray(String[]::new));
+        deleteObjectsResult.getDeletedObjects().addAll(
```

Comment:
here we modify the to-be-returned keys, so this is something that could be verified too (referring to the comment about the unit test)